oap-server/server-bootstrap/src/main/resources/lal/envoy.yaml

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+rules:
+  - name: envoy
+    dsl: |
+      filter {
+        json {
+        }
+        // only collect error logs
+        if (parsed?.response?.responseCode < 500 && !parsed?.commonProperties?.responseFlags) {
```

Comment:
We should set the sampling mechanism separated for responseFlags is null or not null when `parsed?.response?.responseCode < 500`, because upstream service error matters too.

Reply:
Do you want to also sample those logs when `parsed?.response?.responseCode < 500`, like 404, 302?

Reply:
Yes, even there is no internal error.